src/jsn.erl

```diff
@@ -357,6 +359,71 @@ path_elements(_Type, _Path, _Acc) -> erlang:error(badarg).
 
 
 %%==============================================================================
+%% select API
+%%==============================================================================
+
+%%------------------------------------------------------------------------------
+%% @doc Transforms a list of Elements according to a given OutputSpecification
+%%------------------------------------------------------------------------------
+-spec select(selection(), json_array()) -> json_array().
+select(OutputSpec, Elements) when is_list(Elements) ->
+    select(OutputSpec, [], Elements);
+select(_OutpuSpec, _Elements) ->
+    erlang:error(badarg).
+
+%%------------------------------------------------------------------------------
+%% @doc Same as select/2 but it first filters the elements according to a given
+%% list of Conditions
+%%------------------------------------------------------------------------------
+-spec select(selection(), conditions(), json_array()) -> json_array().
+select(OutputSpec, Conditions, Elements) when is_list(Elements) ->
+    %% ensures Conditions are a list
+    ConditionList =
+        case Conditions of
+            _ when is_list(Conditions) -> Conditions;
+            _ -> [Conditions]
+        end,
+    %% This function applies a single condition to a single element
+    ApplyConditionFun = 
+        fun({Path, FilterFun}, E) when is_function(FilterFun) ->
+                erlang:apply(FilterFun, [?MODULE:get(Path, E)]);
+           ({Path, Value}, E) ->
+                ?MODULE:get(Path, E) == Value;
+           (FilterFun, E) when is_function(FilterFun) ->
```

Comment:
One other thing here: use

```erlang
when is_function(FilterFun, 1)
```

here. Since you know it's supposed to be a 1-arity function, and you're calling it with 1 argument, you should guard that.